validator/client/propose.go

```diff
@@ -205,7 +206,15 @@ func ProposeExit(
 	ctx, span := trace.StartSpan(ctx, "validator.ProposeExit")
 	defer span.End()
 
-	signedExit, err := CreateSignedVoluntaryExit(ctx, validatorClient, nodeClient, signer, pubKey)
+	genesisResponse, err := nodeClient.GetGenesis(ctx, &emptypb.Empty{})
+	if err != nil {
+		return errors.Wrap(err, "failed to get genesis")
+	}
+	epoch, err := CurrentEpoch(genesisResponse.GenesisTime)
+	if err != nil {
+		return errors.Wrap(err, "gRPC call to get genesis time failed")
+	}
+	signedExit, err := CreateSignedVoluntaryExit(ctx, validatorClient, nodeClient, signer, pubKey, epoch)
```

Comment:
We are now calling genesis twice here, and still passing the node client in.
instead, i think it would be better to call it once outside, and pass in the epoch and slot , there will need to be a check to see if the slot is in the epoch used though.

Reply:
Actually I take this back the whole point of passing the slot here is to calculate the fork, 
sig, err := signVoluntaryExit(ctx, validatorClient, signer, pubKey, exit, currentSlot)

and it should be in the signing epoch, so you can actually take the first slot of the epoch as the slot and pass it in, you don't even need to calculate the slot again.